0004-dockerised-local-network.md

```diff
@@ -0,0 +1,77 @@
+5. **Testing resources**: The testing will be conducted on a local machine. The testing may also require additional tools for monitoring and debugging.
+
+## Drawbacks
+
+One drawback of this approach is the potential for discrepancies between the behavior of the lightweight Mina local network and the full Mina network. While the lightweight network is designed to be "close to real", it may not perfectly replicate all aspects of the full network, which could lead to misleading test results. To mitigate this drawback, we can state clearly in the documentation that the lightweight network is for local development only and is not intended to be a perfect replica of the full network. Mina local networks do not guarantee production-ready results. Users and developers must always double-check their work against real networks.
+
+Another potential drawback is the need for Docker as a dependency. Users who do not have Docker installed or who are unfamiliar with Docker may face additional hurdles in setting up and using the Mina local network.
+
+## Rationale and alternatives
+
+The Dockerization of Mina local networks is proposed as a solution to the current challenges in deploying and running local Mina networks for testing purposes. By encapsulating the Mina daemon and other necessary components in a Docker image, we can provide a simple, fast, and resource-efficient solution for spinning up local Mina networks.
+
+An alternative to this approach would be to improve the existing methods for deploying and running Mina networks, such as optimizing the setup process or providing better support for cloud-based solutions. However, these alternatives would likely be more complex and time-consuming to implement and would not offer the same level of simplicity and efficiency as the Docker-based solution.
+
+Another future option might be in utilizing [OpenMina](https://openmina.com/) project, which can target `Web` / `JS` runtime and provide a lightweight Mina local network in a browser. This would be a great option for developers, however, it is not available at the moment.
+
+## Prior art
+
+Dockerization is a common practice in software development, particularly for applications that require complex setup processes or that need to run in isolated environments. Many blockchain projects also provide Docker images for their nodes or other components, recognizing the benefits of Docker in terms of simplicity, reproducibility, and resource efficiency.
+
+## Unresolved questions
+
+- What is the best way to handle the building phase? And how can we ensure that the Docker image is kept up-to-date with the latest changes to the Mina Daemon and other components?
+  - Should we add additional Minaprotocol CI job to periodically (say, once a day) build the lightweight Mina Daemon, build Docker image and publish it on behalf of O(1) Labs Docker Hub account?
```

Comment:
I don't think we need to build it on every PR or on every push/merge into main branches since doing so will either overwrite existing images too often or we will end up storing good amount of "legacy" images, in which case the clean-up job will be required.
In addition we don't know the actual demand, what people really expect of it.
Taking above into consideration I'd start simple and just do Nightly builds for now.